src/data/finset/basic.lean

```diff
@@ -2108,6 +2112,14 @@ lemma image_bind_filter_eq [decidable_eq α] (s : finset β) (g : β → α) :
   (s.image g).bind (λa, s.filter $ (λc, g c = a)) = s :=
 bind_filter_eq_of_maps_to (λ x, mem_image_of_mem g)
 
+lemma bind_erase {α β : Type*} [decidable_eq β] (f : α → finset β) (s : finset α) (b : β) :
+  s.bind (λ x, (f x).erase b) = (s.bind f).erase b :=
```

Comment:
I wonder if this would be more helpful in the other direction (possibly even a simp lemma, maybe not though).

Reply:
I did it in this direction to be analogous to `bind_insert`, which pulls out the `insert`.  This seemed useful because this gives you access to the fact that `b` not a member -- I don't have strong thoughts one way or another which direction this lemma should go.

Support for the other way is `inter_bind` and `bind_inter`, which the intersection into the bind expression, and `erase` might be more like intersection than insert.

Reply:
I don't think `bind_insert` is the right comparison here - it pulls out the insert from the left argument of the `bind`, but here you're pulling from the right argument. So in fact I think this is better the other way round, and should be called `erase_bind` instead. `bind_erase` should look more like `(s.erase b).bind f = ...`

Reply:
Given the new name for `bind`, this is now
```
lemma erase_bUnion {α β : Type*} [decidable_eq β] (f : α → finset β) (s : finset α) (b : β) :
  (s.bUnion f).erase b = s.bUnion (λ x, (f x).erase b) :=
begin
  ext y,
  simp only [exists_prop, finset.mem_bUnion, ne.def, finset.mem_erase],
  tauto,
end
```